balboa-client/src/main/java/com/blist/metrics/impl/queue/MetricFileQueue.java

```diff
@@ -72,18 +84,26 @@ private void open() throws IOException {
         this.directory.mkdirs();
 
         String logName = basename + ".data";
-        fileStream = new FileOutputStream(new File(directory, logName), true);
-        stream = new BufferedOutputStream(fileStream);
-        reopenTime = now + MetricQueue$.MODULE$.AGGREGATE_GRANULARITY();
+        file = new File(directory, logName);
+        stream = new BufferedOutputStream(new FileOutputStream(new File(directory, logName), true));
+        reopenTime = now + this.reopenInterval;
         metricCount = 0;
     }
 
+    private void renameFile(File file) {
```

Comment:
Let's make this more descriptive like: renameFileWithImmutableExtension or markFileAsImmutable? 

Nit: Also, can we return the result from the rename from the function and do the logging outside the function?